arm/proofs/aes.ml

```diff
@@ -0,0 +1,393 @@
+  GEN_REWRITE_CONV ONCE_DEPTH_CONV EL_CLAUSES THENC
+  DEPTH_CONV WORD_JOIN_CONV;;
+
+let aes_sub_bytes_select_CONV = REWRITE_CONV [aes_sub_bytes_select]
+  THENC GEN_REWRITE_CONV ONCE_DEPTH_CONV [joined_GF2_CONV `joined_GF2`]
+  THENC TOP_DEPTH_CONV let_CONV
+  THENC DEPTH_CONV (WORD_RED_CONV ORELSEC NUM_RED_CONV);;
+let aes_sub_bytes_CONV = REWRITE_CONV [aes_sub_bytes]
+  THENC aes_sub_bytes_select_CONV
+  THENC DEPTH_CONV (WORD_RED_CONV ORELSEC NUM_RED_CONV);;
+let aes_shift_rows_CONV = REWRITE_CONV [aes_shift_rows]
+  THENC DEPTH_CONV (WORD_RED_CONV ORELSEC NUM_RED_CONV);;
+let FFmul02_CONV = REWRITE_CONV [FFmul02]
+  THENC GEN_REWRITE_CONV ONCE_DEPTH_CONV [joined_FFmul_02_CONV `joined_FFmul_02`]
+  THENC DEPTH_CONV (WORD_RED_CONV ORELSEC NUM_RED_CONV);;
+let FFmul03_CONV = REWRITE_CONV [FFmul03]
+  THENC GEN_REWRITE_CONV ONCE_DEPTH_CONV [joined_FFmul_03_CONV `joined_FFmul_03`]
+  THENC DEPTH_CONV (WORD_RED_CONV ORELSEC NUM_RED_CONV);;
+let aes_mix_word_CONV = REWRITE_CONV [aes_mix_word]
+  THENC FFmul02_CONV
+  THENC FFmul03_CONV
+  THENC DEPTH_CONV (WORD_RED_CONV ORELSEC NUM_RED_CONV);;
+let aes_mix_columns_CONV = REWRITE_CONV [aes_mix_columns]
+  THENC aes_mix_word_CONV
+  THENC TOP_DEPTH_CONV let_CONV
+  THENC DEPTH_CONV (WORD_RED_CONV ORELSEC NUM_RED_CONV);;
+let aese_helper_CONV = REWRITE_CONV [aese]
+  THENC aes_shift_rows_CONV
+  THENC aes_sub_bytes_CONV
+  THENC DEPTH_CONV (WORD_RED_CONV ORELSEC NUM_RED_CONV);;
+let aesmc_helper_CONV = REWRITE_CONV [aesmc]
+  THENC aes_mix_columns_CONV;;
+
+(** Stop early if unmatched. Conversions will become extremely expensive if we don't stop early *)
+let aese_CONV tm =
+    match tm with
+      Comb(Comb(Const("aese",_),
+           Comb(Const("word",_),d)),
+           Comb(Const("word",_),n))
+    when is_numeral d && is_numeral n -> aese_helper_CONV tm
+  | _ -> failwith "aese_CONV: inapplicable";;
+let aesmc_CONV tm =
+    match tm with
+      Comb(Const("aesmc",_), Comb(Const("word",_),n))
+    when is_numeral n -> aesmc_helper_CONV tm
+  | _ -> failwith "aesmc_CONV: inapplicable";;
+
+joined_GF2_CONV `joined_GF2`;;
+(REWRITE_CONV [input] THENC aes_sub_bytes_select_CONV) `aes_sub_bytes_select input 0`;;
+(REWRITE_CONV [input] THENC aes_sub_bytes_CONV) `aes_sub_bytes input`;;
+(REWRITE_CONV [input] THENC aes_shift_rows_CONV) `aes_shift_rows input`;;
+time FFmul02_CONV `FFmul02 (word 0x2a)`;;
+time FFmul03_CONV `FFmul03 (word 0x2a)`;;
+time (REWRITE_CONV [input] THENC aes_mix_word_CONV) `aes_mix_word input 0 8 16 24`;;
+time (REWRITE_CONV [input] THENC aes_mix_columns_CONV) `aes_mix_columns input`;;
+time (REWRITE_CONV [input] THENC aese_CONV) `aese (word 0xae6910a45715645a02502baaf5a826c9) (word 0xec882f3270973907d69635eea82d71)`;;
+time (REWRITE_CONV [input] THENC aesmc_CONV) `aesmc (word 0xae6910a45715645a02502baaf5a826c9)`;;
+
+
+
+(* let aes_inv_sub_bytes = new_definition ``;;
+
+let aes_inv_shift_rows = new_definition ``;;
+
+let aes_inv_mix_columns = new_definition ``;; *)
```

Comment:
What do you think about removing comments here? :)

Reply:
Finished1